### PR TITLE
Introduces scoped search. A search can be scoped to the provided folder.

### PR DESCRIPTION
**Pull request by bonafidehan**
Usage: `dbxcli search query /some-folder`

Manually tested.

### CODE REVIEW

File: cmd/search.go [hunk 1]
```diff
@@ -18,17 +18,29 @@ import (
...
 
 func search(cmd *cobra.Command, args []string) (err error) {
-	if len(args) != 1 {
+	if len(args) == 0 {
 		return errors.New("`search` requires a `query` argument")
 	}
 
-	arg := files.NewSearchArg("", args[0])
+	// Parse path scope, if provided.
+	scope := ""
```
Comment (diwakergupta) [inline cmd/search.go]: [nit] Slight preference for `var scope string`; Go uses the value `""` as default for strings.

File: cmd/search.go [hunk 2]
```diff
@@ -18,17 +18,29 @@ import (
...
+	if len(args) == 2 {
+		scope = args[1]
+		if len(scope) > 0 {
```
Comment (diwakergupta) [inline cmd/search.go]: I'm actually not sure how we'd get `len(args) == 2` with `len(scope) == 0`. Even if we did, the prefix check below will still fail for it. So just remove the outer if?

**Comment by diwakergupta** [comment]
@bonafidehan sorry, you'll have to rebase -- I just pushed some major breaking changes to both the SDK and dbxcli :(

File: cmd/search.go (hunk 2 above)
Comment (bonafidehan) [inline cmd/search.go]: args is `[] string`, while scope is a `string`. But you are right, `HasPrefix` works as expected in degenerate case. Changing.

**Comment by bonafidehan** [comment]
Rebased. Updated based on comments.

**Comment by diwakergupta** [comment]
Yay, thanks @bonafidehan!

**Pull request merged**